pkg/receiver/telegrafreceiver/receiver.go

```diff
@@ -105,6 +110,37 @@ func (r *telegrafreceiver) Start(ctx context.Context, host component.Host) error
 	return err
 }
 
+// Consume metrics and retry on recoverable errors
+func (r *telegrafreceiver) consumeWithRetry(ctx context.Context, metrics pdata.Metrics) error {
+	constantBackoff := backoff.WithMaxRetries(backoff.NewConstantBackOff(r.consumeRetryDelay), r.consumeMaxRetries)
```

Comment:
I wonder if this won't end up slowing this whole thing down. Perhaps we could try reusing those backoffs? 🤔  Are you eager to benchmark this e.g. using a https://pkg.go.dev/sync#Pool ?

Reply:
Right now, we don't even need to be threadsafe, and could just reuse one backoff held in the receiver object.

I haven't benchmarked this, but intuitively it should be quite fast (definitely faster than metrics conversion), and I haven't seen any impact in integration tests. `ConstantBackoff` is a single-field struct, so it should be allocated on the stack: https://github.com/cenkalti/backoff/blob/6b0e4ad0cd65431b217393bff47d1dff727b264b/backoff.go#L57.

Reply:
Fair enough. One `time.Duration` should be lightweight enough to not pay too much attention to this.